File: Structural/Decorator/Cpp/MilkDecorator.cpp
// MilkDecorator.cpp

#include "MilkDecorator.h"

MilkDecorator::MilkDecorator(std::unique_ptr<ICoffee> coffee)
  : CoffeeDecorator(std::move(coffee)) {}

std::string MilkDecorator::getDescription() const {
  return decoratedCoffee->getDescription() + ", Milk";
}

double MilkDecorator::getCost() const {
  return decoratedCoffee->getCost() + 1.5;
}

File: Structural/Decorator/CSharp/MilkDecorator.cs
// MilkDecorator.cs

public class MilkDecorator : CoffeeDecorator {
  public MilkDecorator(ICoffee coffee) : base(coffee) {}

  public override string GetDescription() {
    return decoratedCoffee.GetDescription() + ", Milk";
  }

  public override double GetCost() {
    return decoratedCoffee.GetCost() + 1.5;
  }
}
